src/core/battle.rs

```diff
@@ -43,6 +46,36 @@ pub struct Id(i32);
 #[derive(Serialize, Deserialize, Default, Clone, Copy, Debug, PartialEq, PartialOrd)]
 pub struct Strength(pub i32);
 
+#[derive(Serialize, Deserialize, Clone, Copy, Debug, PartialEq, PartialOrd)]
+pub enum Weight {
+    Normal = 0,
+    Heavy = 1,
+    Immovable = 2,
+}
+impl Default for Weight {
+    fn default() -> Self {
+        Weight::Normal
+    }
+}
+impl fmt::Display for Weight {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+       match *self {
+        Weight::Normal => write!(f, "Normal"),
+        Weight::Heavy => write!(f, "Heavy"),
+        Weight::Immovable => write!(f, "Immovable"),
+       }
+    }
+}
+
+#[derive(Serialize, Deserialize, Default, Clone, Copy, Debug, PartialEq, PartialOrd)]
+pub struct PushStrength(pub Weight);
+
+impl PushStrength {
+    pub fn can_push(self, weight: Weight) -> bool {
+        weight != Weight::Immovable && self.0 <= weight
```

Comment:
Adding logic in this file might not be what's best for the architecture, but I didn't know where to put it.